systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -238,6 +238,49 @@ void testCruiseControlChangesFromRebalancingtoProposalReadyWhenSpecUpdated() {
         KafkaRebalanceUtils.waitForKafkaRebalanceCustomResourceState(clusterOperator.getDeploymentNamespace(), testStorage.getClusterName(), KafkaRebalanceState.ProposalReady);
     }
 
+    @IsolatedTest
+    void testCruiseControlChangesToProposalReadyWhenRestartedWhileRebalancing() {
```

Comment:
I don't think this will work well because you rely on lucky timing too much. You should IMHO need to test it with some unit tests using mocks.

Reply:
It works fine for me. We can also add a unit test using the `MockCruiseControl` class.

Reply:
Added the unit test

Reply:
It is great that you added the unit test. But my point was that you should remove it because this will not be reliable test.